### PR TITLE
Windows build

### PR DESCRIPTION
**Pull request by bam241**
Those change were required to allow the alamagamated version of PyNE used in DAGMC.

Not sure if this will not break the test.

if so I'll tweak this.

### CODE REVIEW

**Comment by bam241** [comment]
@gonuke @pshriwise this should be ready for review.
this is needed to build dagmc on windows.

File: src/data.cpp [hunk 1]
```diff
@@ -1596,15 +1596,18 @@ double pyne::decay_const(std::string nuc) {
...
-    if (result.size() == 1) {
-        return result[0];
-    }
-    return 1.0/0.0;
+  std::vector<double> result = data_access<double, level_data>(
+      nuc, 0.0, DBL_MAX, offsetof(level_data, half_life), level_data_lvl_map);
+  if (result.size() == 1) {
+    return result[0];
+  }
+#ifndef _WIN32
```
Comment (gonuke) [inline src/data.cpp]: We should return the same result, regardless of platform.

File: src/h5wrap.h [hunk 2]
```diff
@@ -230,7 +230,7 @@ namespace h5wrap
     int arr_ndim = H5Sget_simple_extent_dims(arr_space, arr_dims, NULL);
 
     // Read in data from file to memory
-    T mem_arr [arr_dims[0]];
+    T * mem_arr = new T [arr_dims[0]];
```
Comment (gonuke) [inline src/h5wrap.h]:
```suggestion
    T* mem_arr = new T [arr_dims[0]];
```

File: src/source_sampling.cpp [hunk 3]
```diff
@@ -632,14 +632,14 @@ int pyne::Sampler::get_cell_list_size() {
    // 0: for unstructured mesh
    // 1: for sub-voxel R2S
    // max_num_cells: for voxel R2S
-   if (ve_type == moab::MBTET or has_cell_fracs == false) {
-      return 0;
+   if (ve_type == moab::MBTET || has_cell_fracs == false) {
```
Comment (gonuke) [inline src/source_sampling.cpp:635]: Surprised this ever worked ?!?!

File: src/utils.cpp [hunk 4]
```diff
@@ -385,7 +384,11 @@ std::string pyne::get_full_filepath(std::string filename) {
   // remove all extra whitespace
   filename = pyne::remove_characters(" " , filename);
   // use stdlib call
+#ifndef _WIN32
   const char* full_filepath = realpath(filename.c_str(), NULL);
+#else
+  const char* full_filepath = filename.c_str();
```
Comment (gonuke) [inline src/utils.cpp]: This seems like different behavior...?

File: src/material_library.cpp [hunk 5]
```diff
@@ -301,20 +303,18 @@ void pyne::MaterialLibrary::write_hdf5_nucpath(hid_t db,
...
-  int nuc_data[nuc_size];
-  {
-    int n = 0;
-    for (auto nuc : nuclist) {
-      nuc_data[n] = nuc;
-      n++;
-    }
+  std::vector<int> nuc_data;
+  for (auto nuc : nuclist) {
+    nuc_data.push_back(nuc);
```
Comment (gonuke) [inline src/material_library.cpp]: Maybe do as above (line 290)
```suggestion
  nuc_data.assign(nuclist.begin(), nuclist.end())
```

**Review by gonuke — CHANGES_REQUESTED** [review]
There are a couple of places where the windows behavior appears different from non-windows.

File: src/utils.cpp (hunk 4 above)
Comment (bam241) [reply src/utils.cpp]: realpath does not exist on windows.
https://www.quora.com/In-C++-why-isnt-the-function-realpath-on-Windows

Comment (gonuke) [reply src/utils.cpp]: OK - but then should we change the non-windows behavior?  Do we know that `realpath()` is always necessary?

File: src/data.cpp (hunk 1 above)
Comment (gonuke) [reply src/data.cpp]: And we should also make sure that this result is not expected in other places!

File: src/utils.cpp (hunk 4 above)
Comment (bam241) [reply src/utils.cpp]: I defaulted to use `canonical` for windows :https://en.cppreference.com/w/cpp/filesystem/canonical

File: src/data.cpp (hunk 1 above)
Comment (bam241) [reply src/data.cpp]: maybe we could disable half-life for windows ?

Comment (gonuke) [reply src/data.cpp]: Is there no way to return some equivalent of infinity in windows?

**Review by gonuke — APPROVED** [review]

**Pull request merged**